### ISSUE TITLE
Does object_store use VisualStudioCodeCredential?

### ISSUE THREAD

**Issue by thomasfrederikhoeck** [apache/arrow-rs#5534]
**Which part is this question about**:
`object_store`

**Describe your question**
TLDR; Is object store using something like VisualStudioCodeCredential?

I ran into the following in `delta-rs` which uses `object_store`  where we see some weird bug (https://github.com/delta-io/delta-rs/issues/2306). 

It only happens if the code runs inside VS Code and not in a script started from the command line. It only happens for some storage accounts and only for some users. 

It let me to think that it could related to the fact that you (at least in python SDK) can authenticate using VS Code (don't ask med how) https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.visualstudiocodecredential?view=azure-python. Does `object_store` in anyway do the same? 

**Additional context**

**Comment by tustvold** [comment]
We don't explicitly support this, but the documentation is very sparse on the mechanics of how this is configured

**Comment by thomasfrederikhoeck** [comment]
You were exactly right! - thank you @tustvold ! "AZURE_STORAGE_ACCOUNT_KEY" was set in an `.env` file, which resulted in VS Code setting it. And then it appears that `azure_use_azure_cli` is ignored when "AZURE_STORAGE_ACCOUNT_KEY" is set.
